pygfx/objects/_lights.py

```diff
@@ -1,20 +1,27 @@
 import math
 
+import pylinalg as la
+import numpy as np
+
 from ._base import WorldObject
 from ..utils.color import Color
-from ..linalg import Matrix4, Vector3
 from ..cameras import Camera
 from ..resources import Buffer
 from ..cameras import OrthographicCamera, PerspectiveCamera
 from ..utils import array_from_shadertype
 
 
-def get_pos_from_camera_parent_or_target(light):
+def get_pos_from_camera_parent_or_target(light: "Light") -> np.ndarray:
     if isinstance(light.parent, Camera):
-        p = light.position.clone().add(Vector3(0, 0, -1))
-        return p.apply_matrix4(light.parent.matrix_world)
+        cam = light.parent
+        transform = cam.world.matrix
+        return la.vector_apply_matrix((0, 0, -1), transform)
+    elif isinstance(light, SpotLight):
+        return light.target.world.position
+    elif isinstance(light, DirectionalLight):
+        return light.target.world.position
     else:
-        return light.target.get_world_position()
+        raise ValueError("Unknown light source.")
 
 
 class Light(WorldObject):
```

Comment:
Let's set `_FORWARD_IS_MINUS_Z = True` while we're at it. This might change the behavior of the point-light-shadow, because the `look_at` ill-defined-up-mechanism kicks in for some directions.

Reply:
Done.